src/main/java/com/aws/greengrass/clientdevices/auth/iot/Thing.java

```diff
@@ -40,6 +40,11 @@ public final class Thing implements AttributeProvider, Cloneable {
     // map of certificate ID to the time this certificate was known to be attached to the Thing
     private final Map<String, Instant> attachedCertificateIds;
     private boolean modified = false;
+    private final Source source;
+
+    public enum Source {
+        LOCAL, CLOUD, UNSET;
+    }
```

Comment:
feel free to ignore, just a thought:
Can we live _without_ maintaining this `Source` metadata? I see it is used only to distinguish between things stored locally (in registry) and thing attached to the core in the cloud and they are treated the same way everywhere else. Theoretically, with current use case, we only need thing-names of the thing attached to the core. Maintaining a `Source` metadata just to be able to create `Thing` domain object for these attached thing-names seems bit of an overhead.

(rest of the changes look good to me)